spec/rspec/matchers/dsl_spec.rb

```diff
@@ -563,6 +563,28 @@ def foo
         expect(self.matcher.matches?('other value')).to be_falsey
       end
 
+      context "with a negated match block" do
+        let(:matcher) do
+          new_matcher(:name, "value") do |expected|
+            match do |actual|
+              expect(actual).to eq expected
+            end
+
+            match_when_negated do |actual|
+              expect(actual).not_to eq expected
+            end
+          end
+        end
+
+        it "returns true if the wrapped expectation passes" do
+          expect(self.matcher.does_not_match?('other value')).to be_truthy
+        end
```

Comment:
Putting this here if only to track my own thoughts to not get lost in this confusing bug :)
![](http://screenshots.chrisarcand.com/vmree.jpg)